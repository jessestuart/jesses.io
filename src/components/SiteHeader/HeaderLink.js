/* @flow */
import { Link } from 'gatsby'
import React from 'react'

import classNames from 'classnames'

type Props = {
  children: *,
  className?: *,
  href: *,
  isActive: *,
}

const HeaderLink = ({ children, className, href, isActive = false }: Props) => {
  const headerClassNames: string = classNames(
    'lato f4 ph3 mv2 pv2 flex-auto',
    className,
    {
      'white-90 fw4': isActive,
      'light-silver fw3': !isActive,
    }
  )
  return (
    <ul className={headerClassNames}>
      <Link className="shadow-none flex justify-center items-center" to={href}>
        {children}
      </Link>
    </ul>
  )
}

export default HeaderLink
