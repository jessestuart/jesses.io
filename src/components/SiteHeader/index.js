import React from 'react'
import _ from 'lodash'
import Link from 'gatsby-link'

import config from 'gatsby-config'

/**
 * Render a SiteHeader. Size options are 'large' and 'small'. Default to 'small'.
 */
const SiteHeader = () => (
  <h3 className="sans-serif fw2 f3 pa3">
    <Link className="color-inherit no-underline shadow-none" to={'/'}>
      {_.get(config, 'siteMetadata.title')}
    </Link>
  </h3>
)

export default SiteHeader
