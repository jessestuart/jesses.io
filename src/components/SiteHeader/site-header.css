.site-header::before {
  content: '';
  position: absolute;

  /* background-color: red; */

  /* height: 100%; */

  /* width: 4000px; */

  /* left: -2000px; */

  /* z-index: -1; */
}
