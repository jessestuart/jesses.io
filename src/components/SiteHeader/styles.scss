@import '../utils/colors.scss';

.header {
  color: $white;
  height: 60px;
}
