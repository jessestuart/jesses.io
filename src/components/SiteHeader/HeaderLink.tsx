import classNames from 'classnames'
import Link from 'gatsby-link'
import _ from 'lodash'
import React, { ReactNode } from 'react'
import { Text } from 'rebass'
import styled from 'styled-components'
import { ThemeProvider } from 'theme-ui'

import { HeaderTheme } from 'components/SiteHeader/SiteHeader'
import StyledLink from 'components/StyledLink/StyledLink'
import Theme from 'styles/Theme'

const StyledHeaderLink = styled(Link)`
  transition: all 0.5s;
  &:hover {
    color: ${Theme.colors.textLight};
  }
`

interface Props {
  children?: ReactNode
  className?: string
  href: string
  isActive?: boolean
  pathname?: string
}

const getIsActive = ({ href, pathname }): boolean =>
  (href === '/' && pathname === '/') ||
  (href !== '/' && _.startsWith(pathname, href))

const getThemeForPathname = (pathname: string): HeaderTheme =>
  _.startsWith(pathname, '/photography') ? HeaderTheme.LIGHT : HeaderTheme.DARK

const HeaderLink = (props: Props) => {
  const { children, className, href, pathname } = props
  const isActive = getIsActive({ href, pathname })
  const theme = getThemeForPathname(pathname)

  return (
    <StyledHeaderLink
      className={classNames(
        'b--mid-gray f4 fw3 mv2 ph3 pv2 shadow-none',
        {
          'black-90 fw5': isActive && theme === HeaderTheme.LIGHT,
          'light-silver': !isActive && theme === HeaderTheme.LIGHT,
          'light-silver fw3': !isActive && theme === HeaderTheme.DARK,
          'white-90 fw4': isActive && theme === HeaderTheme.DARK,
        },
        className,
      )}
      to={href}
    >
      {children}
    </StyledHeaderLink>
  )
}

export default HeaderLink
