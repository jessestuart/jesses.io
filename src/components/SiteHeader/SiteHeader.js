import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Headroom from 'react-headroom'
import _ from 'lodash'

import HeaderLink from './HeaderLink'

import '../../styles/base.css'

const initialState = { isHamburgerMenuEnabled: false }

class SiteHeader extends Component {
  render() {
    const pathname = _.get(this, 'props.location.pathname')
    const isRoot = _.isNil(pathname) || pathname === '/'
    return (
      <Headroom
        style={{
          MozTransition: 'all 0.7s ease-in-out',
          OTransaction: 'all 0.7s ease-in-out',
          Transition: 'all 0.7s ease-in-out',
          WebkitTranstion: 'all 0.7s ease-in-out',
        }}
      >
        <header
          className={`bb b--hot-pink center site-header flex ${
            !isRoot ? 'bw1' : ''
          }`}
          style={{ background: '#282C34' }}
        >
          <div className="w-35-ns flex flex-row center source-sans fw3">
            <HeaderLink className="br b--mid-gray" href={'/'} isActive={isRoot}>
              home
            </HeaderLink>
            <HeaderLink
              className="br b--mid-gray"
              href={'/posts'}
              isActive={pathname.startsWith('/posts')}
            >
              words
            </HeaderLink>
            <HeaderLink
              className="br b--mid-gray"
              href={'/photography'}
              isActive={pathname.startsWith('/photography')}
            >
              pictures
            </HeaderLink>
            <HeaderLink
              href={'/about'}
              isActive={pathname.startsWith('/about')}
            >
              about
            </HeaderLink>
          </div>
        </header>
      </Headroom>
    )
  }
}

SiteHeader.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
}

SiteHeader.defaultProps = {
  location: {
    pathname: '/',
  },
}

SiteHeader.state = { ...initialState }

export default SiteHeader
