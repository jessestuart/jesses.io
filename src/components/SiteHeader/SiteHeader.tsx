import 'styles/base.css'

import _ from 'lodash'
import React from 'react'
import Headroom from 'react-headroom'

import GatsbyLocation from 'types/GatsbyLocation'

import HeaderLink from './HeaderLink'

interface Props {
  location: GatsbyLocation
}

interface Header {
  title: string
  href: string
}

const Headers: Header[] = [
  { href: '/', title: 'home' },
  { href: '/posts', title: 'words' },
  { href: '/photography', title: 'pictures' },
  { href: '/about', title: 'about' },
]

const SiteHeader = (props: Props) => {
  const pathname: string = _.get(props, 'location.pathname', '/')
  const isRoot: boolean = _.eq(pathname, '/')
  return (
    <Headroom
      className="bg-gray-primary"
      style={{
        MozTransition: 'all 0.7s ease-in-out',
        OTransition: 'all 0.7s ease-in-out',
        WebkitTransition: 'all 0.7s ease-in-out',
        transition: 'all 0.7s ease-in-out',
      }}
    >
      <header
        id="site-header"
        className={`b--hot-pink bb bg-gray-primary center flex pb1 site-header w-100 ${
          !isRoot ? 'bw1' : ''
        }`}
      >
        <ol className="center flex flex-row fw3 mb1 source-sans w-35-ns">
          {Headers.map((header: Header, index: number) => (
            <li key={header.title}>
              <HeaderLink
                className={index === Headers.length - 1 ? '' : 'br b--mid-gray'}
                href={header.href || `/${header.title}`}
                pathname={pathname}
              >
                {header.title}
              </HeaderLink>
            </li>
          ))}
        </ol>
      </header>
    </Headroom>
  )
}

export default SiteHeader
