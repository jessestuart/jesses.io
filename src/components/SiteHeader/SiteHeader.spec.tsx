import React from 'react'
import renderer from 'react-test-renderer'

import SiteHeader from './SiteHeader'

jest.mock('react-headroom', () => 'Headroom')

test('Render SiteHeader component.', () => {
  const location = { location: { pathname: '/' } }
  const component = renderer.create(<SiteHeader location={location} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Render SiteHeader component as not-root.', () => {
  const location = { location: { pathname: '/blog' } }
  const component = renderer.create(<SiteHeader location={location} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
