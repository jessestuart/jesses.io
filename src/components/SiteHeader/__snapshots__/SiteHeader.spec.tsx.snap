// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SiteHeader component as not-root. 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Alegreya Sans;
  font-weight: 700;
  line-height: 1.25;
  font-family: Alegreya Sans;
  line-height: 1.25;
  font-weight: 700;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Lato,system-ui,sans-serif;
  background-color: rgb(55,59,70);
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c0 {
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
}

<Headroom
  className="c0"
>
  <header
    className="c1 bb flex b--hot-pink"
    fontFamily="body"
    id="site-header"
  >
    <div
      className="c2 gray-primary fw3 w-35-ns"
    >
      <a
        className="c3 f4 fw3 mv2 ph3 pv2 shadow-none white-90 fw4 b--mid-gray br b--mid-gray"
        href="/true"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        home
      </a>
      <a
        className="c3 f4 fw3 mv2 ph3 pv2 shadow-none light-silver b--mid-gray br b--mid-gray"
        href="/true/posts"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        words
      </a>
      <a
        className="c3 f4 fw3 mv2 ph3 pv2 shadow-none light-silver b--mid-gray br b--mid-gray"
        href="/true/photography"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        pictures
      </a>
      <a
        className="c3 f4 fw3 mv2 ph3 pv2 shadow-none light-silver b--mid-gray"
        href="/true/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        about
      </a>
    </div>
  </header>
</Headroom>
`;

exports[`Render SiteHeader component. 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Alegreya Sans;
  font-weight: 700;
  line-height: 1.25;
  font-family: Alegreya Sans;
  line-height: 1.25;
  font-weight: 700;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Lato,system-ui,sans-serif;
  background-color: rgb(55,59,70);
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c0 {
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
}

<Headroom
  className="c0"
>
  <header
    className="c1 bb flex b--hot-pink"
    fontFamily="body"
    id="site-header"
  >
    <div
      className="c2 gray-primary fw3 w-35-ns"
    >
      <a
        className="c3 f4 fw3 mv2 ph3 pv2 shadow-none white-90 fw4 b--mid-gray br b--mid-gray"
        href="/true"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        home
      </a>
      <a
        className="c3 f4 fw3 mv2 ph3 pv2 shadow-none light-silver b--mid-gray br b--mid-gray"
        href="/true/posts"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        words
      </a>
      <a
        className="c3 f4 fw3 mv2 ph3 pv2 shadow-none light-silver b--mid-gray br b--mid-gray"
        href="/true/photography"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        pictures
      </a>
      <a
        className="c3 f4 fw3 mv2 ph3 pv2 shadow-none light-silver b--mid-gray"
        href="/true/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        about
      </a>
    </div>
  </header>
</Headroom>
`;
